R/utilities-mapping.R

```diff
@@ -133,11 +134,30 @@ getDefaultCaptions <- function(data, metaData = NULL, variableList = colnames(da
 #' Creates default legend captions subset
 #' @keywords internal
 .asLegendCaptionSubset <- function(labels, unit = NULL) {
+
   # Keep ordering of labels as is if factor
-  captionLevels <- sort(unique(labels))
   if (isOfType(labels, "factor")) {
     captionLevels <- levels(labels)
+  } else {
+    captionLevels <- sort(unique(labels))
   }
+
+  # If group name is longer than charactersWidth, then it will be wrapped on
+  # several lines of charactersWidth lenght and cut on non word character.
+  charactersWidth <- 50
```

Comment:
This wrapping is only for residual text elements that do not use `ggtext::element_textbox()` (legend and ticks labels).  It serves the same purpose than  `font$maxWidth` but instead of being a graphical unit, it is a maximum number of characters. 
I like reusing the maxWidth setting, according to my experiments, I think we can set it this way: charactersWidth = maxWidth / 2

Reply:
Could you try with different font size ?
Since maxWidth uses pt as unit, I think it may also be a function of size.

Otherwise, I am okay with maxWidth/2.
That way, users (=me in Reporting Engine) should be able to set up line breaks for legends and tick labels

Reply:
The problem I'm facing is that the part of the code that takes care of wraping data labels (+ turning them into factor) has no connection with theming (this happens in a Group object). So in the end I decided to setup a tlfEnv variable which was the easier/fastest way to implement this and in line with yesterday's Yuri email.